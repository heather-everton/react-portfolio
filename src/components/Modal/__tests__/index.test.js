// __tests__/Modal.test.js with hard coded categories
import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModal = jest.fn();
const currentPhoto = {
  name: 'Weather Dashboard',
  category: 'projects',
  description: "Get a quiuck overview of today's weather and the upcoming forecast.",
  url: 'https://heather-everton.github.io/Weather-Dashboard/',
  github: 'https://github.com/heather-everton/Weather-Dashboard',
  index: 1
};

afterEach(cleanup);

describe('Modal component', () => {
  it('renders', () => {
    render(<Modal
      onClose={mockToggleModal}
      currentPhoto={currentPhoto}
    />);
  })

  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Modal
      onClose={mockToggleModal}
      currentPhoto={currentPhoto}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
})

describe('Click Event', () => {
  it('calls onClose handler', () => {
    const { getByText } = render(<Modal
      onClose={mockToggleModal}
      currentPhoto={currentPhoto}
    />);
    fireEvent.click(getByText('Close this modal'))

    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
})  