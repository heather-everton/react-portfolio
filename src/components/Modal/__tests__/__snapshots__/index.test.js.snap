// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Weather Dashboard 
      </h3>
      <a
        href="https://heather-everton.github.io/Weather-Dashboard/"
        rel="noreferrer"
        target="_blank"
      >
        Check it Out!
      </a>
      <img
        alt="current category"
      />
      <p>
        Get a quiuck overview of today's weather and the upcoming forecast.
      </p>
      <a
        href="https://github.com/heather-everton/Weather-Dashboard"
        rel="noreferrer"
        target="_blank"
      >
        Github Repository
      </a>
      <br />
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
